docs/chapters/incremental_indexing/index.md

```diff
@@ -0,0 +1,29 @@
+# Incremental Indexing in Jina
+
+## Generating Documents Ids:
+
+In Jina, the document ID is by default generated deterministically by computing a hash the document content. This avoids repeating the doc id while reindexing by using a Random or Incremental counter. The user can `override` document ids with Jina client and add `tags` to map to their unique concepts.
+
+## Incremental Indexing Design Pattern:
+
+The rationale behind Incremental indexing is to avoid re-indexing previously indexed documents, thus saving time and compute. This is done by persisting the document identifiers of indexed docs and using them as a DocIdCache.
```

Comment:
I would emphatize that it is a possibility, and not comes out of the box with any indexer

Reply:
I would clarify that if someone does not expect to send the same document again it can work without adding the extra caches, but that if they want to avoid duplicating documents, they need to use one of these patterns.

Reply:
On the other hand, the problem is not only that u may reindex and you will save computation, the problem is it won't work (That's what we were seeing in the `hackathon`)